File: 1046/main.cpp
class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
        priority_queue<int> pq;
        for (auto x : stones)
            pq.push(x);

        while (pq.size() >= 2) {
            int a = pq.top(); pq.pop();
            int b = pq.top(); pq.pop();

            if (a == b)
                continue;
            else
                pq.push(a - b);
        }

        if (pq.empty())
            return 0;
        else
            return pq.top();
    }
};

File: 1046/main.java
class Solution {
	public int lastStoneWeight(int[] stones) {
		PriorityQueue<Integer> pq = new PriorityQueue<>(31, Collections.reverseOrder());
		for (int x : stones)
			pq.add(x);

		while (pq.size() >= 2) {
			int a = pq.poll();
			int b = pq.poll();

			if (a == b)
				continue;
			else
				pq.add(a - b);
		}

		if (pq.size() == 0)
			return 0;
		else
			return pq.peek();
	}
}